iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/IcebergTableUtil.java

```diff
@@ -106,4 +111,61 @@ public static PartitionSpec spec(Configuration configuration, Schema schema) {
     });
     return builder.build();
   }
+
+  public static void updateSpec(Configuration configuration, Table table) {
+    // get the new partition transform spec
+    PartitionSpec newPartitionSpec = spec(configuration, table.schema());
+    if (newPartitionSpec == null) {
+      LOG.debug("Iceberg Partition spec is not updated due to empty partition spec definition.");
```

Comment:
Is this scenario possible? Would this happen if you ran `ALTER TABLE tbl SET PARTITION SPEC()`? The parser would not complain about this?

Reply:
We can get null values if the SessionState is somehow lost. Or the method is called from a different process (like Tez). I just wanted to make sure we log this.